tests/test_assembly.py

```diff
@@ -279,6 +577,229 @@ def test_save_raises(nested_assy):
         nested_assy.save("nested.step", "DXF")
 
 
+@pytest.mark.parametrize(
+    "assy_fixture, count",
+    [("simple_assy", 3), ("nested_assy", 3), ("empty_top_assy", 1),],
+)
+def test_leaf_node_count(assy_fixture, count, request):
+
+    assy = request.getfixturevalue(assy_fixture)
+    _, doc = toCAF(assy, True)
+
+    assert len(get_doc_nodes(doc, True)) == count
+
+
+@pytest.mark.parametrize(
+    "assy_fixture, expected",
+    [
+        (
+            "chassis0_assy",
+            [
+                (
+                    ["chassis", "wheel-axle.*", "wheel:.*"],
+                    {
+                        "color": (1.0, 0.0, 0.0, 1.0),
+                        "color_shape": (1.0, 0.0, 0.0, 1.0),
+                        "num_nodes": 4,
+                    },
+                ),
+                (
+                    ["chassis", "wheel-axle.*", "wheel:.*", "wheel.*_part"],
+                    {"color": (1.0, 0.0, 0.0, 1.0), "num_nodes": 4},
+                ),
+                (
+                    ["chassis", "wheel-axle.*", "axle"],
+                    {
+                        "color": (0.0, 1.0, 0.0, 1.0),
+                        "color_shape": (0.0, 1.0, 0.0, 1.0),
+                        "num_nodes": 2,
+                    },
+                ),
+                (
+                    ["chassis", "wheel-axle.*", "axle", "axle_part"],
+                    {"color": (0.0, 1.0, 0.0, 1.0), "num_nodes": 2},
+                ),
+            ],
+        ),
+    ],
+)
+def test_colors_assy0(assy_fixture, expected, request):
+    """Validate assembly colors with document explorer.
+
+    Check toCAF wth color shape parameter False.
+    """
+
+    def check_nodes(doc, expected):
+        expected = copy.deepcopy(expected)
```

Comment:
I'm not sure I understand the purpose of this `deepcopy` here.

Reply:
You are right.  It can be omitted here in `test_colors_assy0`.  I had copied the function from `test_colors_assy1`.